### ISSUE TITLE
New target state prevented from being applied, when network without both subnet and gateway created

### ISSUE THREAD

**Issue by anujdeshpande** [balena-os/balena-supervisor#1693]

**Comment by pipex** [comment]
Here is the replication

Create the following docker-compose and perform a push in local mode. You'll get a looping exception with the message "Network IPAM config entries must have both a subnet and gateway", and will not be able to push a correct target state after that.

```
version: '2.1'

services:
  hello:
    build: ./hello-world
    networks:
      my_net:
        ipv4_address: 172.16.238.10

networks:
  my_net:
    driver: bridge
    ipam:
      driver: default
      config: 
        - gateway: 172.16.238.1
```

**Comment by pipex** [comment]
The PR #1656 fixed validation for network ipam config, checking that both network and subnet are defined for each ipam config entry (as described in the [docker documentation](https://docs.docker.com/compose/compose-file/compose-file-v2/#networks)) 

The old was querying the wrong parts of the configuration making the validation never to happen [here](https://github.com/balena-os/balena-supervisor/pull/1656/files#diff-f9a66d4ba0463ef433190b8c8c5b5dd77886557585cd0c6d01f27f0398ad1f63R217).

After that PR, the validations throws an exception if the network target state is incorrect, but this turns out to be the wrong approach, because that exception is also triggered when querying target state. 

This isn't a problem in normal operation, but it is in local mode, because local mode queries the old target state before sending a new one. Since the query fails, the CLI can never push the new target state.

I think the solution is just to show a warning on the supervisor logs, since a poorly configured network will not cause the engine to misbehave, it will only render the container unreachable through the provided address.

**Comment by 20k-ultra** [comment]
> I think the solution is just to show a warning on the supervisor logs, since a poorly configured network will not cause the engine to misbehave, it will only render the container unreachable through the provided address.

I think this is a product question @pipex. We totally should prevent the Supervisor from becoming unresponsive as the original reporter in the jellyfish link mentioned but don't know if logging a misconfigured network but continuing to apply target state is better then throwing the exception and not continuing to apply the target state.

**Comment by pipex** [comment]
@20k-ultra  I totally agree that the supervisor should be able to reject a target state, which [it does](https://github.com/balena-os/balena-supervisor/blob/master/src/device-state.ts#L51-L53) and we should work on improving that part of the codebase (better validation, surfacing errors to the dashboard, etc.).

But I also think is clear that the validation cannot happen in the way is being done now, as an exception when building the object, because that will break other parts of the codebase. 

I think to remove this bug, a warning is enough for now, particularly since this should not cause any critical failures, and we can create an issue about better validation and surfacing errors in target state.

**Comment by 20k-ultra** [comment]
the PR is merged, you didn't want it to auto close this ? The version is available in production so maybe we can ? @pipex

**Comment by 20k-ultra** [comment]
closing because resolved in https://github.com/balena-os/balena-supervisor/pull/1694 which is available in Supervisor v12.6.8
